src/controllers/controller_execute.hpp

```diff
@@ -65,7 +65,7 @@ Result controller_execute(std::vector<Circuit> &input_circs,
   //    i is the instruction index in the experiment
   //    j is the param index in the instruction
   //    pars = [par0, par1, ...] is a list of different parameterizations
-  using pos_t = std::pair<uint_t, uint_t>;
+  using pos_t = std::pair<int_t, int_t>;
```

Comment:
I think it is not necessary to change `unit_t` to `int_t` to point this params are global phases, and use pre-defined value such as `GLOBAL_PHASE_POS` instead of using `-1`